app/account/profile/page.tsx

```diff
@@ -0,0 +1,516 @@
+import logECS from '@/utils/clientLogger';
+import styles from './profile.module.scss';
+
+const ProfilePage: React.FC = () => {
+  const router = useRouter();
+  const [otherField, setOtherField] = useState(false);
+  // We need to save the original data for when users cancel their form updates
+  const [originalData, setOriginalData] = useState<ProfileDataInterface>();
+  const [formData, setFormData] = useState<ProfileDataInterface>({
+    firstName: '',
+    lastName: '',
+    affiliationName: '',
+    affiliationId: '',
+    otherInstitution: '',
+    languageId: '',
+    languageName: '',
+  })
+  const [isEditing, setIsEditing] = useState(false);
+  // Errors returned from request
+  const [errors, setErrors] = useState<string[]>([]);
+  // Client-side validation field errors
+  const [fieldErrors, setFieldErrors] = useState<FormErrorsInterface>({
+    firstName: '',
+    lastName: '',
+    affiliationName: '',
+    affiliationId: '',
+    languageId: '',
+    languageName: '',
+    otherInstitution: ''
+  });
+  const [emailAddresses, setEmailAddresses] = useState<EmailInterface[]>([]);
+  const [languages, setLanguages] = useState<LanguageInterface[]>([]);
+  const [, forceRender] = useState(false);
+
+  // Initialize user profile mutation
+  const [updateUserProfileMutation, { loading: updateUserProfileLoading }] = useUpdateUserProfileMutation();
+
+  // Run queries
+  const { data: languageData, error: languageError, refetch: languageRefetch } = useLanguagesQuery();
+  const { data, loading: queryLoading, error: queryError, refetch } = useMeQuery();
+
+  // Client-side validation of fields
+  const validateField = (name: string, value: string) => {
+    let error = '';
+    switch (name) {
+      case 'firstName':
+        if (!value || value.length <= 2) {
+          error = `Please enter a valid name.`;
+        }
+        break;
+      case 'lastName':
+        if (!value || value.length <= 2) {
+          error = `Please enter a valid name.`;
+        }
+        break;
+      case 'affiliationName':
+        if (!value || value.length <= 2) {
+          error = 'Institution name cannot be blank.';
+        }
+        break;
+    }
+    setFieldErrors(prevErrors => ({
+      ...prevErrors,
+      [name]: error
+    }));
+    return error;
+  }
+
+  const profileUpdateMutation = async () => {
+    const response = await updateUserProfileMutation({
+      variables: {
+        input: {
+          givenName: formData.firstName,
+          surName: formData.lastName,
+          affiliationId: formData.affiliationId,
+          languageId: formData.languageId,
+        }
+      },
+    });
+    return response.data;
+  }
+  // Update Profile info
+  const updateProfile = async () => {
+    try {
+      const response = await profileUpdateMutation();
+      if (response) {
+        setIsEditing(false);
+      }
+    } catch (error) {
+      if (error instanceof ApolloError) {
+        await handleApolloErrors(
```

Comment:
You're probably wondering why I have to call "handleApolloErrors" on these mutations, rather than updating the apollo-wrapper `errorLink` to do the same thing. I have tried several times to try and just use the `errorLink` with the same logic as in `handleApolloErrors()`, but things appear to get out of sync when I do that. 

I'll have to experiment more with it, but so far have not yet been successful with seamless functionality.